### PR TITLE
fix: comments votes

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes https://github.com/decred/politeiagui/issues/2467

### Solution description

This PR does 3 things:

1. Make sure votes are sorted by date (older ones first).
2. Calculate user score by `commentid` and save it in `commentsLikes`. For example:

```
{
  1: 0,
  2: -1.
  3: 1.
  4: -1,
  5: 1
  ...
}
```

Before we were just making a `unionBy commentid` and storing it in the state which was causing data to be lost.

3. Rename `RECEIVE_SYNC_LIKE_COMMENT` to `RECEIVE_LIKE_COMMENT ` and cleanup some old code.

### UI Changes Screenshot

https://user-images.githubusercontent.com/13955303/124768942-7546a680-df0f-11eb-9483-a0bd935868fc.mp4

### CODE REVIEW

**Review by thi4go — COMMENTED** [review]
Hey mate, code looks good, bugs are fixed. For documenting purposes, this PR also fixes a bug where the UI would highlight a thumb even when the like/dislike was undone:

image

@lukebp while testing the PR, I found that if I tried to spam the like button, `politeiad` would return me a 500 error, which I think needs to be handled gracefully:

```
2021-07-07 11:56:55.555 [ERR] POLI: 127.0.0.1:60395 POST /v2/pluginwrite HTTP/1.1 Internal error 1625669815: handlePluginWrite: PluginWrite: commentVoteSave: duplicate blob
2021-07-07 11:56:55.555 [ERR] POLI: Stacktrace (NOT A REAL CRASH): goroutine 867 [running]:
runtime/debug.Stack(0xc00007ca00, 0x19c2601, 0x21)
	/usr/local/Cellar/go/1.16.5/libexec/src/runtime/debug/stack.go:24 +0x9f
main.respondWithErrorV2(0x1c05a20, 0xc0002de000, 0xc0005fe800, 0x19c3f39, 0x22, 0x1bf6580, 0xc000d963b0)
	/Users/thiagofigueiredo/go/src/github.com/decred/politeiacopy/politeiad/v2.go:943 +0x15b7
main.(*politeia).handlePluginWrite(0xc0002cbc80, 0x1c05a20, 0xc0002de000, 0xc0005fe800)
	/Users/thiagofigueiredo/go/src/github.com/decred/politeiacopy/politeiad/v2.go:509 +0x6f5
main.logging.func1(0x1c05a20, 0xc0002de000, 0xc0005fe800)
	/Users/thiagofigueiredo/go/src/github.com/decred/politeiacopy/politeiad/politeiad.go:142 +0x25e
main.closeBody.func1(0x1c05a20, 0xc0002de000, 0xc0005fe800)
	/Users/thiagofigueiredo/go/src/github.com/decred/politeiacopy/politeiad/politeiad.go:149 +0x43
net/http.HandlerFunc.ServeHTTP(0xc0005f9ab0, 0x1c05a20, 0xc0002de000, 0xc0005fe800)
	/usr/local/Cellar/go/1.16.5/libexec/src/net/http/server.go:2069 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0005f0300, 0x1c05a20, 0xc0002de000, 0xc0005fe600)
	/Users/thiagofigueiredo/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xd3
net/http.serverHandler.ServeHTTP(0xc000308540, 0x1c05a20, 0xc0002de000, 0xc0005fe600)
	/usr/local/Cellar/go/1.16.5/libexec/src/net/http/server.go:2887 +0xa3
net/http.(*conn).serve(0xc000be8be0, 0x1c07cc0, 0xc0002ca140)
	/usr/local/Cellar/go/1.16.5/libexec/src/net/http/server.go:1952 +0x8cd
created by net/http.(*Server).Serve
	/usr/local/Cellar/go/1.16.5/libexec/src/net/http/server.go:3013 +0x39b
```
Increasing the debounce time prevents this from happening, but also gives very bad UX since the UI takes more time to visually show your action succeeded.

**Comment by lukebp** [comment]
@thi4go you are correct.  I opened an [issue](https://github.com/decred/politeia/issues/1455) for the backend bug.

**Review by thi4go — APPROVED** [review]
tACK

**Comment by tiagoalvesdulce** [comment]
Yes, the bugs are related. Both were caused by the way we were handling `/comments/votes`.

**Review by victorgcramos — APPROVED** [review]
LGTM

File: src/reducers/models/comments.js [hunk 1]
```diff
@@ -127,20 +139,13 @@ const comments = (state = DEFAULT_STATE, action) =>
             };
 
             return compose(
-              set(["commentsLikes", "backup"], backupForCommentLikes),
               set(["commentsLikes", "byToken", shortToken], newCommentsLikes),
-              set(["comments", "backup"], comments),
+              // This will cause some delay to update up/downvotes if there are a lot of comments in the page. This happens because we have to recreate the comments array all the time due to the redux immutability pattern. Will be improved when we migrate to the plugins structure.
```
Comment (victorgcramos) [inline src/reducers/models/comments.js:143]: well explained. Should we open an issue to track it?

Comment (amassarwi) [reply src/reducers/models/comments.js:143]: agreed, it's too wide tho, would limit it to 80 chars columns.

**Review by lukebp — APPROVED** [review]
ACK

**Pull request merged**